? Makefile
? OctreeSceneManager.diff
? include/Makefile
? include/OgreTerrainSceneManager.diff
? include/OgreTerrainSceneManager.h.diff
? src/Makefile
Index: include/Makefile.am
===================================================================
RCS file: /cvsroot/ogre/ogrenew/PlugIns/OctreeSceneManager/include/Makefile.am,v
retrieving revision 1.5
diff -u -3 -r1.5 Makefile.am
--- include/Makefile.am	16 Jun 2004 20:55:59 -0000	1.5
+++ include/Makefile.am	29 Sep 2004 21:25:37 -0000
@@ -1,4 +1,4 @@
-noinst_HEADERS = \
+pkginclude_HEADERS = \
 		OgreHeightmapTerrainPageSource.h \
 		OgreOctreeCamera.h \
         OgreOctreeNode.h \
Index: include/OgreTerrainSceneManager.h
===================================================================
RCS file: /cvsroot/ogre/ogrenew/PlugIns/OctreeSceneManager/include/OgreTerrainSceneManager.h,v
retrieving revision 1.18
diff -u -3 -r1.18 OgreTerrainSceneManager.h
--- include/OgreTerrainSceneManager.h	27 Aug 2004 23:08:40 -0000	1.18
+++ include/OgreTerrainSceneManager.h	29 Sep 2004 21:25:38 -0000
@@ -207,12 +207,12 @@
     /** Returns the TerrainRenderable that contains the given pt.
         If no tile exists at the point, it returns 0;
     */
-    TerrainRenderable * getTerrainTile( const Vector3 & pt );
+    virtual TerrainRenderable * getTerrainTile( const Vector3 & pt );
 
     /** Returns the TerrainPage that contains the given pt.
     If no page exists at the point, it returns 0;
     */
-    TerrainPage* getTerrainPage( const Vector3 & pt );
+    virtual TerrainPage* getTerrainPage( const Vector3 & pt );
 
     /** Creates a RaySceneQuery for this scene manager. 
     @remarks
Index: src/OgreTerrainRenderable.cpp
===================================================================
RCS file: /cvsroot/ogre/ogrenew/PlugIns/OctreeSceneManager/src/OgreTerrainRenderable.cpp,v
retrieving revision 1.44
diff -u -3 -r1.44 OgreTerrainRenderable.cpp
--- src/OgreTerrainRenderable.cpp	14 Jul 2004 02:47:17 -0000	1.44
+++ src/OgreTerrainRenderable.cpp	29 Sep 2004 21:25:39 -0000
@@ -248,6 +248,9 @@
         _calculateMinLevelDist2( C );
         if (msOptions->lit)
             _calculateNormals();
+	    
+        if (msOptions->coloured)
+		_generateVertexLighting(Vector3(0,500,100000), ColourValue(0.7,0.7,0.7));
 
     }
     //-----------------------------------------------------------------------
@@ -326,7 +329,7 @@
 
 
         Vector3 cpos = cam -> getDerivedPosition();
-        Vector3 diff = mCenter - cpos;
+        Vector3 diff = (this->getWorldPosition() + mCenter) - cpos;
 
         Real L = diff.squaredLength();
 
